extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -481,13 +487,20 @@ export class AmpA4A extends AMP.BaseElement {
           // src cache issue.  If we decide to keep a SafeFrame-like solution,
           // we should restructure the promise chain to pass this info along
           // more cleanly, without use of an object variable outside the chain.
+          if (!creativeParts) {
+            return Promise.resolve();
+          }
           if (this.experimentalNonAmpCreativeRenderMethod_ !=
               XORIGIN_MODE.CLIENT_CACHE &&
-              creativeParts &&
               creativeParts.creative) {
             this.creativeBody_ = creativeParts.creative;
           }
-          if (!creativeParts || !creativeParts.signature) {
+          if (creativeParts.size && creativeParts.size.length == 2) {
+            // We'll need to remember the creative size for the x-domain case.
+            this.creativeSize_ = creativeParts.size;
+            this.handleResize(creativeParts.size[0], creativeParts.size[1]);
```

Comment:
Actually, calling it a second time doesn't make sense and isn't necessary, so I've removed that. Since we call handleResize as soon as we have the creative's size, that's the best we can do, and we don't need to do anything after that.